react-native/index.d.ts

```diff
@@ -4471,11 +4471,15 @@ declare module "react" {
 
         type Style = ViewStyle | TextStyle | ImageStyle
 
+        type NamedStyles<T> = {
+            [P in keyof T]: Style;
+        }
+
         /**
          * Creates a StyleSheet style reference from the given object.
          */
-        export function create<T>( styles: T ): T;
-
+        export function create<T extends NamedStyles<T>>( styles: T ): T;
```

Comment:
@mhegazy Don't think so. That results in `T` being inferred as `{}`, e.g. given

```ts
const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
    }
});
```

Type of `styles` is now `{}` and hence all uses of `styles.container` are type errors.

I should have added an extra test as the existing ones don't catch this problem - essentially what we get from `react-native init AwesomeProject` with the index file changed to `.tsx` and the import fixed to use `* as React`.